addons/hw_drivers/tools/upgrade.py

```diff
@@ -0,0 +1,205 @@
+"""Module to manage odoo code upgrades using git"""
+
+import logging
+import platform
+import requests
+import subprocess
+from functools import cache, wraps
+from odoo.addons.hw_drivers.tools.helpers import path_file, require_db, writable, odoo_restart, toggleable
+
+_logger = logging.getLogger(__name__)
+
+
+@cache
+def is_linux():
+    """Check if the current system is Linux.
+
+    :return: True if the system is Linux, False otherwise
+    """
+    return platform.system() == 'Linux'
```

Comment:
why having a function for this ? it would make more sense to be a constant IMHO, something like:
`IS_IOT_BOX = platform.system() == 'Linux'`